qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/DesiredNodesUpgradeIT.java

```diff
@@ -50,9 +50,11 @@ private enum ProcessorsPrecision {
     public void testUpgradeDesiredNodes() throws Exception {
         assumeTrue("Desired nodes was introduced in 8.1", getOldClusterVersion().onOrAfter(Version.V_8_1_0));
 
-        if (getOldClusterVersion().onOrAfter(Processors.DOUBLE_PROCESSORS_SUPPORT_VERSION)) {
+        var featureVersions = new MetadataFeatures().getHistoricalFeatures();
+
+        if (getOldClusterVersion().onOrAfter(featureVersions.get(DesiredNode.DOUBLE_PROCESSORS_SUPPORTED))) {
```

Comment:
This will need to wait until we work out how to do bwc tests for serverless features

Reply:
++